internal/cluster/cluster.go

```diff
@@ -1098,6 +1181,18 @@ func (m *Manager) setConnectivityMajorityGroupsForClusterInternal(cluster *commo
 	return nil
 }
 
+func (m *Manager) DetectAndStoreCollidingIPsForCluster(clusterID strfmt.UUID, db *gorm.DB) error {
```

Comment:
Should be done only in monitoring, so this function is not needed

Reply:
I also agree that this should be done in monitoring. But please also note that in the current context of the inventory you are retrieving the cluster and hosts twice because they were already retrieved in the calling function. In addition, the calling function of this flow is invoked only in interactive mode but I think that the collision should be a general validation (not only in SaaS)